ntbs-service/Pages/ContactDetails/Index.cshtml.cs

```diff
@@ -9,17 +12,41 @@ namespace ntbs_service.Pages.ContactDetails
     public class IndexModel : PageModel
     {
         private readonly IUserService _userService;
-
-        public IndexModel(IUserService userService)
+        private readonly IReferenceDataRepository _userRepository;
+        
+        public IndexModel(IUserService userService, IReferenceDataRepository userRepository)
         {
             _userService = userService;
+            _userRepository = userRepository;
         }
         
         public User ContactDetails { get; set; }
         
+        [BindProperty(SupportsGet = true)]
+        public string Username { get; set; }
+        
         public async Task<IActionResult> OnGetAsync()
         {
-            ContactDetails = await _userService.GetUser(User);
+            if (Username == null)
+            {
+                ContactDetails = await _userService.GetUser(User);
+                ViewData["IsEditable"] = true;
+            }
+            else
+            {
+                ContactDetails = await _userRepository.GetCaseManagerByUsernameAsync(Username);
+            }
+            
+
+            if (ContactDetails == null)
+            {
+                return NotFound();
+            }
+
+            if (!StringValues.IsNullOrEmpty(Request.Headers["Referer"]))
+            {
+                ViewData["Referer"] = Request.Headers["Referer"].ToString();
```

Comment:
Or, actually, why don't we pick a fallback at this stage? Then the view doesn't have to worry about it :)